LFS-178 As a user, I can record tumors data

prabnur opened the pull request.

prabnur committed:
LFS-168 / LFS-178 Generate clinical data questionnaires for Tumors Data

prabnur committed:
LFS-168 / LFS-178 Remove an empty key that is unnecessary

prabnur committed:
LFS-168 / LFS-178 Refactor JSON file

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+  },
+  "Age at diagnosis (y)": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Age at diagnosis (y)",
```
This should use the new `unitOfMeasurement` option from #119

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "minAnswers": 0,
+    "maxAnswers": 1,
+    "displayMode": "input",
+    "dataType": "vocabulary"
```
If it's a `vocabulary` question, `sourceVocabulary` must also be given.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+  },
+  "Topography of primary lesion": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Topography of primary lesion",
```
It feels like we should have the user use SNOMED for topographies -- what do you think, @marta- ?

marta- replied:
@OrdiNeu I temporarily removed all vocabularies other than HPO. We'll rework this part when we have installed vocabularies working as expected.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "maxAnswers": 1,
+    "displayMode": "list",
+    "dataType": "boolean",
+    "Yes": {
```
These aren't required for a `boolean` question -- instead, either use `yesLabel`, `noLabel`, and `unknownLabel`, or (in this case where it's exactly Yes/No/Unknown) just ensure `enableUnknown` is `true`.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "maxAnswers": 1,
+    "displayMode": "list",
+    "dataType": "boolean",
+    "Yes": {
```
As before, just use `enableUnknown`

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "maxAnswers": 1,
+    "displayMode": "list",
+    "dataType": "boolean",
+    "Yes": {
```
`enableUnknown` again. This also applies for the five or six questions below here.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "maxAnswers": 1,
+    "displayMode": "input",
+    "dataType": "vocabulary",
+    "sourceVocabulary": "ICD"
```
I can't find this vocabulary on BioOntology -- do you know which morphology vocabulary this is referring to, @marta- ?

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+  },
+  "Source of Tumor Diagnosis- Pathology report Clinic note Patient report Other (specify) blank= pre requesting..": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Source of Tumor Diagnosis- Pathology report Clinic note Patient report Other (specify) blank= pre requesting..",
```
Please check the grammar of this sentence.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "text": "Stage",
+    "minAnswers": 0,
+    "maxAnswers": 1,
+    "displayMode": "list+input",
```
Shouldn't this just be `list`? What other options are there?

marta- replied:
Apparently stages are different for some brain(?) cancers, and I-IV covers the majority of cancers but not all.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "minAnswers": 0,
+    "maxAnswers": 1,
+    "displayMode": "input",
+    "dataType": "text"
```
This and the below also feels like they should be `vocabulary` questions, but I'm unsure of what vocabulary to use.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+  },
+  "Risk Group Assignment": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Risk Group Assignment",
```
Missing `displayMode`.

marta- replied:
Display mode should be specified if we want to override the default for that data type. I've actually removed it everywhere for simplicity. Do you know of any problems that this may cause?

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "text": "Group/Stage/Risk Classification Schema",
+    "minAnswers": 0,
+    "maxAnswers": 1,
+    "displayMode": "list+input",
```
Just an `input` here -- there's no list given to choose from.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "text": "Symptoms",
+    "minAnswers": 0,
+    "maxAnswers": 0,
+    "displayMode": "list+input",
```
As before, just an `input` here.

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+    "maxAnswers": 1,
+    "displayMode": "input",
+    "dataType": "text"
+  },
```
Why the lack of capitalization here?

OrdiNeu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,547 @@
+  },
+  "Any treatment at PMH?": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Any treatment at PMH?",
```
This sounds like a `boolean` question instead of a `text` one.

OrdiNeu committed:
LFS-178: As a user, I can record tumors data
Refactor so that the .json file parses correctly

OrdiNeu committed:
LFS-178: As a user, I can record tumors data
Fix boolean question configuration

marta- committed:
[code style] Replaced tabs with double spaces

marta- committed:
LFS-178 As a user, I can record tumors data

Work in progress.

- Removed questions regarding patient id, which will be pre-filled based on the linked subject id
- Renamed question field ids
- Removed unnecessary displayType where it matched the default
- Cleaned up question texts (rephrased, capitalized, removed answer format and unit of measurement from text).

marta- committed:
LFS-178 As a user, I can record tumors data
- Temporarily changed dataType from vocabulary to text where vocabularies are not supported yet
- Cleand up question medatata (displayMode, min/maxAnswers, answer ids)

marta- committed:
LFS-178 As a user, I can record tumors data
- Clearer text for the pmh treatments question

marta- reviewed (approved):
Works fine after the latest fixes.

sdumitriu commented on questionnaires/clinical-data/src/main/resources/SLING-INF/content/Tumors.json:
```diff
@@ -0,0 +1,423 @@
+  },
+  "location": {
+    "jcr:primaryType": "lfs:Question",
+    "text": "Location (code)",
```
What kind of `code` is it?

marta- replied:
Custom coding afaik.
